pkg/yang/types.go

```diff
@@ -280,7 +280,8 @@ check:
 		y.IdentityBase = resolvedBase.Identity
 	}
 
-	if t.Range != nil {
+	// If parsing fraction-digits of type decimal64 fails, the range should not be parsed
+	if t.Range != nil && (!isDecimal64 || (isDecimal64 && y.FractionDigits != 0)) {
```

Comment:
`parseRanges` already error checks where `FractionDigits` is 0 (via `decimalValueFromString`), so I'm wondering if it's actually better to keep the current behaviour and error out, instead of silently allowing parsing to continue?

Reply:
https://github.com/openconfig/goyang/blob/fd7d014c61f43a05d53146582e61503f374787df/pkg/yang/types.go#L247-L250
In fact, the value 0 indicates that the FractionDigits value has already failed to be parsed (via `asRangeInt`).

 If continue to parse, `parserRanges` checks the error value 0 as FractionDigits, then reports an unreasonable error msg `bad range: invalid number of fraction digits 0: out of range [1..18]`,  and the new testcase also fails because two errors are generated, so I think the value 0 needs to be excluded.

Reply:
Thanks! Makes sense.